CONVENTIONS.md

```diff
@@ -264,6 +264,9 @@ Please note that in case of [Single Node OpenShift](https://github.com/openshift
   * Two replicas
   * Hard pod [anti-affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#always-co-located-in-the-same-node) on hostname (no two pods on the same node)
   * Use the maxUnavailable rollout strategy on deployments (prefer 25% by default for a value)
+  * If the the operator or operand requires persistent storage
+    * Prefer to use topology-aware volume provisioning, i.e. prefer using a StorageClass with `volumeBindingMode: WaitForFirstConsumer` so that volume binding is delayed until the pod is scheduled (taking anit-affinty into account)
```

Comment:
In the monitoring operator case, it can't really check or enforce that the requested storage class specifies `olumeBindingMode: WaitForFirstConsumer`. Are we saying that the operator should do something (warn? fail?) when it detects that the condition isn't fulfilled?

Reply:
Yeah I wasn't sure how to phrase this. Or maybe this should be added somewhere else?
From what I understand, we want to inform users (who have control over which `StorageClass` is used) what impact their choices can have and what works best with the HA conventions.

Maybe its better to note here that storage HA is left to the storage provider (like we do in the [documentation](https://docs.openshift.com/container-platform/4.9/storage/understanding-persistent-storage.html) already)? I think it would still be useful to outline the possible effects. This seems like a good place in terms of visibility, but I agree its somewhat apart from the rest of the content.

Reply:
> From what I understand, we want to inform users (who have control over which StorageClass is used) what impact their choices can have and what works best with the HA conventions.

This document is targeted towards OpenShift developers, not users. In practice, someone should be able to configure the monitoring operands with static storage resources if their storage provider doesn't support `WaitForFirstConsumer` binding.

Reply:
Yeah I was trying to phrase if openly enough, but point taken. What do you think of the current iteration?